cli/src/config.rs

```diff
@@ -163,26 +163,39 @@ impl AsMut<StackedConfig> for RawConfig {
 }
 
 #[derive(Clone, Debug)]
-enum ConfigPath {
-    /// Existing config file path.
-    Existing(PathBuf),
-    /// Could not find any config file, but a new file can be created at the
-    /// specified location.
-    New(PathBuf),
+enum ConfigPathState {
+    New,
+    Exists,
+}
+
+/// A ConfigPath can be in one of two states:
+///
+/// - exists(): a config file exists at the path
+/// - !exists(): a config file doesn't exist here, but a new file _can_ be
+///   created at this path
+#[derive(Clone, Debug)]
+pub struct ConfigPath {
+    path: PathBuf,
+    state: ConfigPathState,
```

Comment:
Let's keep it private. I don't think users of `ConfigEnv` should be aware of `ConfigPath`, which is implementation detail.